src/components/Plants/DashboardChart.tsx

```diff
@@ -23,13 +23,36 @@ export interface DashboardChartProps {
   minHeight?: string;
 }
 
-export default function DashboardChart(props: DashboardChartProps): JSX.Element {
-  const { chartId, chartLabels, chartValues, minHeight } = props;
+export default function DashboardChart(props: DashboardChartProps): JSX.Element | null {
+  const { chartLabels, chartValues } = props;
+  const { loadedStringsForLocale } = useLocalization();
+  const [locale, setLocale] = useState<string | null>(null);
+
+  useEffect(() => {
+    setLocale(loadedStringsForLocale);
+  }, [loadedStringsForLocale]);
+
+  if (!locale || !chartLabels || !chartValues) {
+    return null;
+  }
+
+  return <DashboardChartContent {...props} locale={locale} chartLabels={chartLabels} chartValues={chartValues} />;
+}
+
+interface DashboardChartContentProps {
+  chartId: string;
+  chartLabels: string[];
+  chartValues: number[];
+  minHeight?: string;
+  locale: string;
+}
+
+function DashboardChartContent(props: DashboardChartContentProps): JSX.Element {
```

Comment:
moved the original chart into a separate component that requires a locale and the values/labels..
the parent component/wrapper now just returns null if they are not set.
this handles most cases of useEffect concurrency issues where chart was being constructed for same canvas multiple times in parallel